### ISSUE TITLE
src/entrybleed.c is dedicated to x86(-64)

### ISSUE THREAD

**Issue by jserv** [bcoles/kasld#3]
The `README.md` claims the support for non-x86 architectures. However, file `src/entrybleed.c` is dedicated to x86(-64) since it relies on inlined assembly: (in function `sidechannel`)
```c
  uint64_t a, b, c, d;
  // Note: clang does not support this asm with intel_syntax noprofix
  __asm__ volatile(".intel_syntax noprefix;"
                   "mfence;"
                   "rdtscp;"
                   "mov %0, rax;"
                   "mov %1, rdx;"
                   "xor rax, rax;"
                   "lfence;"
                   "prefetchnta qword ptr [%4];"
                   "prefetcht2 qword ptr [%4];"
                   "xor rax, rax;"
                   "lfence;"
                   "rdtscp;"
                   "mov %2, rax;"
                   "mov %3, rdx;"
                   "mfence;"
                   ".att_syntax;"
                   : "=r"(a), "=r"(b), "=r"(c), "=r"(d)
                   : "r"(addr)
                   : "rax", "rbx", "rcx", "rdx");
```

**Comment by bcoles** [comment]
Correct. The [Entrybleed](https://www.willsroot.io/2022/12/entrybleed.html) side-channel attack works only Intel x86_64 CPUs and AMD x86_64 CPUs with kPTI disabled (pti=off).

https://github.com/bcoles/kasld/blob/19534d156f5b18255a91dd6e37acc6326e9b87b2/src/entrybleed.c#L1-L27

The `main()` function simply returns `0` if `entrybleed.c` is compiled on an incompatible system.

https://github.com/bcoles/kasld/blob/19534d156f5b18255a91dd6e37acc6326e9b87b2/src/entrybleed.c#L398-L409

**Comment by jserv** [comment]
> Correct. The [Entrybleed](https://www.willsroot.io/2022/12/entrybleed.html) side-channel attack works only Intel x86_64 CPUs and AMD x86_64 CPUs with kPTI disabled (pti=off).

Should the compilation of file src/entrybleed.c be ignored by the build system?

**Comment by bcoles** [comment]
> > Correct. The [Entrybleed](https://www.willsroot.io/2022/12/entrybleed.html) side-channel attack works only Intel x86_64 CPUs and AMD x86_64 CPUs with kPTI disabled (pti=off).
> 
> Should the compilation of file src/entrybleed.c be ignored by the build system?

Ideally, yes it should be ignored. I've moved the platform check to the start of the file in 47caa3e6c7f7f42b6bf0228eac942c8f8238a43c.

The [build system](https://github.com/bcoles/kasld/blob/master/Makefile) naively attempts to build everything in `./src/`.

Compiler errors for individual files are treated as non-fatal. Build failures for individual files are expected when compiling on-target (most often due to missing headers).

Files in `./src/` opt-out when compiling on an incompatible platform through the use of C pre-processor macros. This is a temporary solution until I decided how I want to structure the build system. I would like to support cross-compiling while keeping the project modular.
